File: Diagrammatize.Design.Patterns/code/cxx/Bridge/A3/CharDisplayImpl.cpp
#include "CharDisplayImpl.hpp"

#include <iostream>

CharDisplayImpl::CharDisplayImpl(char head_, char body_, char foot_): head(head_), body(body_), foot(foot_) {
}

void CharDisplayImpl::rawOpen() {
    std::cout << head;
}

void CharDisplayImpl::rawPrint() {
    std::cout << body;
}

void CharDisplayImpl::rawClose() {
    std::cout << foot << std::endl;
}

File: Diagrammatize.Design.Patterns/code/python/Bridge/A3/CharDisplayImpl.py
from DisplayImpl import DisplayImpl

class CharDisplayImpl(DisplayImpl):
    def __init__(self, head, body, foot):
        self.head = head
        self.body = body
        self.foot = foot

    def rawOpen(self):
        print(self.head, end="")

    def rawPrint(self):
        print(self.body, end="")

    def rawClose(self):
        print(self.foot)
